Sources/NIOCore/AsyncAwaitSupport.swift

```diff
@@ -330,3 +330,13 @@ struct AsyncSequenceFromIterator<AsyncIterator: AsyncIteratorProtocol>: AsyncSeq
         self.iterator
     }
 }
+
+@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+extension EventLoop {
+    @inlinable
+    public func makeFutureWithTask<Return>(_ body: @Sendable @escaping () async throws -> Return) -> EventLoopFuture<Return> {
```

Comment:
Can we call it `performWithTask`?

Reply:
I don’t think we should add this extension on ELG: it’s rare that you’re without an opinion as to what loop your promise should be attached to. For that same reason, we don’t offer a `makePromise` API on ELG.

Reply:
OK, sounds reasonable. Any thoughts on the name?

Reply:
I probably have a slight preference for the name proposed by @weissi, as it aligns with existing prior art for making Futures.